src/librustc_typeck/diagnostics.rs

```diff
@@ -1915,6 +1915,47 @@ More details can be found in [RFC 438].
 [RFC 438]: https://github.com/rust-lang/rfcs/pull/438
 "##,
 
+E0182: r##"
+You bound an associated type in an expression path which is not
+allowed. Erroneous code example:
+
+```compile_fail,E0182
+trait Foo {
+    type A;
+    fn bar() -> isize;
+}
+
+impl Foo for isize {
+    type A = usize;
+    fn bar() -> isize { 42 }
+}
+
+pub fn main() {
```

Comment:
Yes. The point is to avoid any unneeded "noise". Everyone knows that code needs to go inside a function and it's not what we're trying to do in here. In the generated output, everything (or almost) is put inside a main function so don't worry about it.